rotkehlchen/chain/evm/decoding/spark/savings/decoder.py

```diff
@@ -121,9 +122,12 @@ def _decode_spark_tokens_deposit_withdrawal(self, context: DecoderContext) -> De
         ):
             return DEFAULT_DECODING_OUTPUT
 
-        vault_token = self.base.get_or_create_evm_token(
-            address=context.tx_log.address,
-            encounter=(encounter := TokenEncounterInfo(should_notify=False)),
+        GlobalDBHandler.set_tokens_protocol_if_missing(
+            tokens=[vault_token := self.base.get_or_create_evm_token(
+                address=context.tx_log.address,
+                encounter=(encounter := TokenEncounterInfo(should_notify=False)),
```

Comment:
are we sure that this is always a spark token? i fear that we assign the protocol to something that is not. Can you check in a test for this flow also that the protocol is set correctly?

Reply:
yes, we're certain it's always a spark savings vault token because this function is only called through the `addresses_to_decoders` mapping in the decoder.

we handle the protocol assignment here because we call `get_or_create_evm_token` at this point in the flow nonetheless. i've also added a test to verify that the protocol gets set correctly for this specific case.

conversation in discord: https://discord.com/channels/657906918408585217/1080783409640648765/1403421129820999741